presto-main/src/main/java/com/facebook/presto/server/security/LdapFilter.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.security;
+
+import com.google.common.base.Splitter;
+import com.google.common.base.Throwables;
+import com.google.common.net.HttpHeaders;
+import io.airlift.log.Logger;
+
+import javax.inject.Inject;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.security.Principal;
+import java.util.Base64;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Objects;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static javax.naming.Context.INITIAL_CONTEXT_FACTORY;
+import static javax.naming.Context.PROVIDER_URL;
+import static javax.naming.Context.SECURITY_AUTHENTICATION;
+import static javax.naming.Context.SECURITY_CREDENTIALS;
+import static javax.naming.Context.SECURITY_PRINCIPAL;
+
+public class LdapFilter
+        implements Filter
+{
+    private static final Logger LOG = Logger.get(LdapFilter.class);
+    public static final String AUTHENTICATION_TYPE = "basic";
+    public static final String LDAP_CONTEXT_FACTORY = "com.sun.jndi.ldap.LdapCtxFactory";
+    private final String ldapBaseDn;
+    private final String ldapUrl;
+
+    @Inject
+    public LdapFilter(LdapServerConfig config)
+    {
+        ldapBaseDn = requireNonNull(config.getBaseDn(), "ldap.base.dn is null");
+        ldapUrl = requireNonNull(config.getLdapUrl(), "ldap.url is null");
+
+        Hashtable<String, String> environment = new Hashtable<>();
+        environment.put(INITIAL_CONTEXT_FACTORY, LDAP_CONTEXT_FACTORY);
+        environment.put(PROVIDER_URL, ldapUrl);
+        try {
+            DirContext context = new InitialDirContext(environment);
```

Comment:
ya.I did the validation here the same way as SnegoFilter does. The purpose of it could be to fail fast in such scenarios. If the ldap server is not reachable, then the clients connecting to the server is going to fail anyway.